hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncTable.java

```diff
@@ -344,6 +345,26 @@ default CheckAndMutateBuilder ifEquals(byte[] value) {
     CompletableFuture<Boolean> thenMutate(RowMutations mutation);
   }
 
+  /**
+   * Batch checkAndMutates the specified data into the table.
+   *
+   * @param checkAndMutates The list of rows to apply.
+   * @return A list of {@link CompletableFuture}s that represent the result for each
+   *   CheckAndMutate.
+   */
+  List<CompletableFuture<Boolean>> checkAndMutate(List<CheckAndMutate> checkAndMutates);
```

Comment:
Can we deprecate old checkAndMutate in favor of this?

Reply:
I think we can do that. If we do so, I will add checkAndMutate method that accepts a single CheckAndMutate object.

@Apache9 What do you think?